tests/unit/Segments/DynamicSegments/FilterDataMapperTest.php

```diff
@@ -166,4 +167,35 @@ public function testItChecksWooCommerceProductId() {
       'action' => WooCommerceProduct::ACTION_PRODUCT,
     ]);
   }
+
+  public function testItMapsWooCommerceNumberOfOrders() {
```

Comment:
You wrote that you are going to create some acceptance tests. I would recommend writing unit tests and integration tests and test your logic in those. They are faster and acceptance tests tend to be more flakey and take more time. Have you heard about [testing pyramid](https://3fxtqy18kygf3on3bu39kh93-wpengine.netdna-ssl.com/wp-content/uploads/2020/01/test-automation-pyramid.jpg)?

I would write just a single acceptance test and more integration tests.

Reply:
Yeah, I'm familiar with the testing pyramid. I agree with your general recommendation of writing more unit tests and integration tests than acceptance tests because of the reasons you laid out. As I mentioned in the PR description, I was planning to write only acceptance tests for this segment because, unless I'm missing something, the other WooCommerce segments contain mostly just acceptance tests, so I was planning to follow this pattern.

I will look into adding unit and integration tests for this new segment.